### ISSUE TITLE
method_missing cop

### ISSUE THREAD

**Issue by connorshea** [bbatsov/rubocop#3104]
Per the [Ruby Style Guide](https://github.com/bbatsov/ruby-style-guide#no-method-missing), overriding `method_missing` should be avoided.

Below is the text from the Ruby Style Guide:

--------

Avoid using `method_missing` for metaprogramming because backtraces become messy, the behavior is not listed in `#methods`, and misspelled method calls might silently work, e.g. `nukes.launch_state = false`. Consider using `delegation`, `proxy`, or `define_method` instead. If you must use `method_missing`:

- Be sure to also define `respond_to_missing?`
- Only catch methods with a well-defined prefix, such as `find_by_*` -- make your code as assertive as possible.
- Call `super` at the end of your statement.
- Delegate to assertive, non-magical methods:

```ruby
# bad
def method_missing?(meth, *params, &block)
  if /^find_by_(?<prop>.*)/ =~ meth
    # ... lots of code to do a find_by
  else
    super
  end
end

# good
def method_missing?(meth, *params, &block)
  if /^find_by_(?<prop>.*)/ =~ meth
    find_by(prop, *params, &block)
  else
    super
  end
end

# best of all, though, would to define_method as each findable attribute is declared
```

--------

I'd like to have a cop that:
- Warns on `def method_missing`.
- Warns on `method_missing` with no `respond_to_missing?` (optional?)
- Warns when `method_missing` doesn't have `super` (optional?)

More resources on `method_missing`, problems with overriding it, and its best practices:
- https://www.alfajango.com/blog/method_missing-a-rubyists-beautiful-mistress/
- http://technicalpickles.com/posts/using-method_missing-and-respond_to-to-create-dynamic-methods/
- https://robots.thoughtbot.com/always-define-respond-to-missing-when-overriding

**Comment by bbatsov** [comment]
That's a good idea. PR welcome! :-)

**Comment by connorshea** [comment]
👏 thanks @haziqhafizuddin!
